Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,309 @@
+
+            // Insert pxmesh into the collider component
+            PhysX::MeshColliderComponentRequestsBus::Event(entityId, &PhysX::MeshColliderComponentRequests::SetMeshAsset, assetId);
+            entity->Deactivate();
+
+            return;
+        }
+
+        entity->CreateComponent<PhysX::EditorShapeColliderComponent>();
+        switch (geometry->type)
+        {
+        case urdf::Geometry::SPHERE:
+            {
+                entity->CreateComponent(LmbrCentral::EditorSphereShapeComponentTypeId);
+                auto sphereGeometry = std::dynamic_pointer_cast<urdf::Sphere>(geometry);
+                entity->Activate();
+                LmbrCentral::SphereShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::SphereShapeComponentRequests::SetRadius, sphereGeometry->radius);
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::BOX:
+            {
+                entity->CreateComponent(LmbrCentral::EditorBoxShapeComponentTypeId);
+                auto boxGeometry = std::dynamic_pointer_cast<urdf::Box>(geometry);
+                entity->Activate();
+                LmbrCentral::BoxShapeComponentRequestsBus::Event(
+                    entityId,
+                    &LmbrCentral::BoxShapeComponentRequests::SetBoxDimensions,
+                    URDF::TypeConversions::ConvertVector3(boxGeometry->dim));
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::CYLINDER:
+            {
+                entity->CreateComponent(LmbrCentral::EditorCylinderShapeComponentTypeId);
+                auto cylinderGeometry = std::dynamic_pointer_cast<urdf::Cylinder>(geometry);
+                entity->Activate();
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetHeight, cylinderGeometry->length);
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetRadius, cylinderGeometry->radius);
+                entity->Deactivate();
+            }
+            break;
+        default:
+            AZ_Warning("AddCollider", false, "Unsupported collider geometry type, %d", geometry->type);
+            break;
+        }
+    }
+    AZStd::string CollidersMaker::GetFullURDFMeshPath(const AZStd::basic_string<char>& modelPath, const AZStd::basic_string<char>& meshPath)
+    {
+        AZStd::string azPath{ modelPath };
+        AZ::IO::BasicPath modelBasicPath(modelPath);
+        modelBasicPath = modelBasicPath.RemoveFilename();
+
+        AZStd::string azMeshPath{ meshPath };
+        AZ::StringFunc::Replace(azMeshPath, "package://", "", true, true);
+        modelBasicPath.Append(azMeshPath);
+
+        return modelBasicPath.String();
```

Comment:
Why accept an `AZStd::basic_string<char>` while an `AZStd::string` type alias can be used.
Also it would be better to accept both parameters as an AZ::I::Path and to return an AZ::IO::Path

```suggestion
    AZ::IO::Path CollidersMaker::GetFullURDFMeshPath(AZ::IO::Path modelPath, AZ::IO::Path meshPath)
    {
        modelPath.RemoveFilename();

        AZ::StringFunc::Replace(meshPath.Native(), "package://", "", true, true);
        modelBasicPath /= meshPath;

        return modelBasicPath;
```